net/golioth/Kconfig

```diff
@@ -323,6 +323,23 @@ config GOLIOTH_SYSTEM_SETTINGS
 	  When selected, Golioth credentials will be loaded from settings
 	  subsystem.
 
+if GOLIOTH_SYSTEM_SETTINGS
```

Comment:
I like that there's an ASSERT to detect values that are too long:
```
/home/mike/golioth-zephyr-workspace/modules/lib/golioth/net/golioth/system_client.c:38:1: error: static assertion failed: "PSK exceeds mbedTLS configured maximum PSK length"
   38 | BUILD_ASSERT(sizeof(TLS_PSK) - 1 <= CONFIG_MBEDTLS_PSK_MAX_LEN,
      | ^~~~~~~~~~~~
```

However, selecting `GOLIOTH_SYSTEM_SETTINGS` means compiled-in credentials can't access the new length settings.

```
config GOLIOTH_SYSTEM_CLIENT_PSK_ID
	string "PSK ID"
	depends on !GOLIOTH_SYSTEM_SETTINGS
```